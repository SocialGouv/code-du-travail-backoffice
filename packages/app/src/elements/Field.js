import styled from "@emotion/styled";
import PropTypes from "prop-types";
import React from "react";
import { Flex, Text } from "rebass";

const Container = styled(Flex)`
  margin-bottom: 0.5rem;
`;

const Error = styled(Text)`
  color: red;
  font-size: 0.75rem;
  font-weight: 600;
  height: 1rem;
`;

const Field = ({ children, error = null, ...props }) => (
  <Container {...props}>
    {children}
    {error !== null && <Error>{error}</Error>}
  </Container>
);

Field.propTypes = {
  children: PropTypes.node,
  error: PropTypes.string,
};

export default Field;
