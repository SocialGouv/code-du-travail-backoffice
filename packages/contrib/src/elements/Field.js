import { Flex } from "rebass";
import styled from "styled-components";

export default styled(Flex)`
  margin-bottom: 0.5rem;
`;
