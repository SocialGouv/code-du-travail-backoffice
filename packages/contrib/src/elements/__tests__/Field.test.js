import React from "react";
import { render } from "@testing-library/react";
import Field from "../Field";

describe("[Contrib] elements/<Field />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Field />);

    expect(container).toMatchSnapshot();
  });
});
