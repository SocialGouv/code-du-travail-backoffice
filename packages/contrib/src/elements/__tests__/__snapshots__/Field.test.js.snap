// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] elements/<Field /> should pass 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0.5rem;
}

<div
  className="emotion-0 emotion-1"
/>
`;

exports[`[Contrib] elements/<Field /> should pass with {error} 1`] = `
.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0.5rem;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: red;
  font-size: 0.75rem;
  font-weight: 600;
  height: 1rem;
}

<div
  className="emotion-2 emotion-3"
>
  <div
    className="emotion-0 emotion-1"
  >
    An Error
  </div>
</div>
`;
